src/libraries/System.Memory/src/System/Buffers/ReadOnlySequence.Helpers.cs

```diff
@@ -248,7 +248,7 @@ internal SequencePosition Seek(long offset, ExceptionArgument exceptionArgument
                 int currentLength = startSegment.Memory.Length - startIndex;
 
                 // Position in start segment, defer to single segment seek
-                if (currentLength > offset)
+                if (currentLength > offset || offset == 0)
```

Comment:
Why is this change needed?

Reply:
When Seek was called with the offset 0, it used SeekMultiSegment.
SeekMultiSegment will loop until it finds a segment with data, then return a SequencePosition with this segment.
So we ended up with a SequencePosition that point to the first nonempty segment, while `.Start` point to the first Segment.

Reply:
Why is this specific to offset == 0? It's not the case any time currentLength == offset?

Reply:
When `currentLength == offset` it point to the first element of the next segment, not the current.
Except when it's 0, because we want to point to the start of the sequence.
But I just noticed that because of this, it's difficult to have SequencePosition pointing to the first byte of the Sequence.